News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/NewFeedActivity.java

```diff
@@ -219,9 +213,32 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
 
         if (requestCode == REQUEST_CODE_OPML_IMPORT && resultCode == RESULT_OK) {
-            String filePath = data.getStringExtra(FilePickerActivity.RESULT_FILE_PATH);
-
-            AsyncTaskHelper.StartAsyncTask(new ImportOpmlSubscriptionsTask(filePath, NewFeedActivity.this));
+            final Uri importUri = data.getData();
+
+            switch (importUri.getScheme()) {
+                case ContentResolver.SCHEME_CONTENT:
+                case ContentResolver.SCHEME_FILE:
+                    new Thread(() -> {
+                        try {
+                            final File cacheFile = new File(getCacheDir().getAbsolutePath() + "/import.opml");
+                            final FileOutputStream outputStream = new FileOutputStream(cacheFile);
+                            byte[] buffer = new byte[4096];
+                            final InputStream inputStream = getContentResolver().openInputStream(importUri);
+
+                            int count;
+                            while ((count = inputStream.read(buffer)) > 0) {
+                                outputStream.write(buffer, 0, count);
+                            }
+                            runOnUiThread(() -> AsyncTaskHelper.StartAsyncTask(new ImportOpmlSubscriptionsTask(cacheFile.getAbsolutePath(), NewFeedActivity.this)));
```

Comment:
Don't we need to close the input- and output-streams here? 🤔

Reply:
Probably... maybe we can use a try-with-resources?

Reply:
Interesting! Never used try-with-resources before! Sounds like the perfect use-case!  👍